src/statsmodels/contrasts.jl

```diff
@@ -0,0 +1,233 @@
+## Specify contrasts for coding categorical data in model matrix. Contrasts types
+## are a subtype of AbstractContrasts. ContrastsMatrix types hold a contrast
+## matrix, levels, and term names and provide the interface for creating model
+## matrix columns and coefficient names.
+##
+## Contrasts types themselves can be instantiated to provide containers for
+## contrast settings (currently, just the base level).
+##
+## ModelFrame will hold a Dict{Symbol, ContrastsMatrix} that maps column
+## names to contrasts.
+##
+## ModelMatrix will check this dict when evaluating terms, falling back to a
+## default for any categorical data without a specified contrast.
+##
+## TODO: implement contrast types in Formula/Terms
+##
+## Dave Kleinschmidt 2015-2016
```

Comment:
While I really appreciate your contribution, I don't think we usually keep the contributor names in the file, as this creates unnecessary churn everytime somebody contributes to a file. git is here for that. :-)